update-agent/core/src/version_map.rs

```diff
@@ -137,7 +121,7 @@ impl VersionMap {
         }
     }
 
-    pub fn get_component(&self, name: &str) -> Option<&ComponentInfo> {
+    pub fn get_component(&self, name: &str) -> Option<&SlotVersion> {
```

Comment:
perhaps it should be renamed to slot_version()? for two reasons:
* the type is SlotVersion, not ComponentInfo anymore
* rust getters [don't usually](https://rust-lang.github.io/api-guidelines/naming.html#c-getter) use the get_ prefix.

Reply:
Thank you for the review. I've renamed it in a separate commit.